src/swap/hw-app-swap/Swap.js

```diff
@@ -0,0 +1,211 @@
+        break;
+    }
+    throw new Error("Swap application report error " + errorMessage);
+  }
+
+  async startNewTransaction(): Promise<string> {
+    let result: Buffer = await this.transport.send(
+      0xe0,
+      START_NEW_TRANSACTION_COMMAND,
+      0x00,
+      0x00,
+      Buffer.alloc(0),
+      this.allowedStatuses
+    );
+    if (!this.isSuccess(result)) this.mapProtocolError(result);
+    if (result.length !== 12)
+      throw new Error("APDU response length should be 12");
+    return result.toString("ascii", 0, 10);
+  }
+
+  async setPartnerKey(partnerNameAndPublicKey: Buffer): Promise<void> {
+    let result: Buffer = await this.transport.send(
+      0xe0,
+      SET_PARTNER_KEY_COMMAND,
+      0x00,
+      0x00,
+      partnerNameAndPublicKey,
+      this.allowedStatuses
+    );
+    if (!this.isSuccess(result)) this.mapProtocolError(result);
+  }
+
+  async checkPartner(signatureOfPartnerData: Buffer): Promise<void> {
+    let result: Buffer = await this.transport.send(
+      0xe0,
+      CHECK_PARTNER_COMMAND,
+      0x00,
+      0x00,
+      signatureOfPartnerData,
+      this.allowedStatuses
+    );
+    if (!this.isSuccess(result)) this.mapProtocolError(result);
+  }
+
+  async processTransaction(transaction: Buffer): Promise<void> {
+    let result: Buffer = await this.transport.send(
+      0xe0,
+      PROCESS_TRANSACTION_RESPONSE,
+      0x00,
+      0x00,
+      transaction,
+      this.allowedStatuses
+    );
+    if (!this.isSuccess(result)) this.mapProtocolError(result);
+  }
+
+  async checkTransactionSignature(transactionSignature: Buffer): Promise<void> {
+    let result: Buffer = await this.transport.send(
+      0xe0,
+      CHECK_TRANSACTION_SIGNATURE,
+      0x00,
+      0x00,
+      transactionSignature,
+      this.allowedStatuses
+    );
+    if (!this.isSuccess(result)) this.mapProtocolError(result);
+  }
+
+  async checkPayoutAddress(
+    payoutCurrencyConfig: Buffer,
+    currencyConfigSignature: Buffer,
+    addressParameters: Buffer
+  ): Promise<void> {
+    if (payoutCurrencyConfig.length > 255) {
+      throw new Error("Currency config is too big");
+    }
+    if (
+      currencyConfigSignature.length < 70 ||
+      currencyConfigSignature.length > 73
+    ) {
+      throw new Error(
+        "Signature should be DER serialized and have length in [70, 73] bytes"
+      );
+    }
+    if (addressParameters.length > 255) {
+      throw new Error("Address parameters is too big");
+    }
+    const bufferToSend: Buffer = Buffer.concat([
+      Buffer.from([payoutCurrencyConfig.length]),
+      payoutCurrencyConfig,
+      currencyConfigSignature,
+      Buffer.from([addressParameters.length]),
+      addressParameters
+    ]);
+    let result: Buffer = await this.transport.send(
+      0xe0,
+      CHECK_PAYOUT_ADDRESS,
+      0x00,
+      0x00,
+      bufferToSend,
+      this.allowedStatuses
+    );
+    if (!this.isSuccess(result)) this.mapProtocolError(result);
```

Comment:
ok so the point of the **default** behavior of `send(` is to do that for you. it makes me thing we do not need `mapProtocolError` idea at all?